hotfixes

bfioca-openai opened the pull request.

bfioca-openai commented on src/app/App.tsx:
```diff
@@ -395,26 +434,32 @@ function App() {
-            if (!textContent) return;
+            // Skip moderation guardrail notice messages altogether
+            // This gets added by the SDK but we don't want to present it in the UI
+            if (textContent.includes('Failed Guardrail Reason: moderation_guardrail')) return;
```
This isn't ideal, but I'm stuck trying to find another way of identifying these messages as they come back from the SDK. cc: @dkundel-openai

nm-openai commented:
Thanks for reverting to the existing UI, But it looks like the guardrails are still causing issues. What is this guardrail_tripped event? Is that supported in the SDK?

In the prior implementation the intent was for the guardrail to clearly run and show the result, and the user could easily implement interruption/escalation/update response logic if it tripped as they'd like - what's the intent here?

<img width="1004" alt="image" src="https://github.com/user-attachments/assets/57bb6f23-59f2-4d50-9932-333932a076f8" />

<img width="757" alt="image" src="https://github.com/user-attachments/assets/4e65c6f1-8670-4a3c-9cab-11ae6c6f6ff8" />

bfioca-openai commented:
> Thanks for reverting to the existing UI, But it looks like the guardrails are still causing issues. What is this guardrail_tripped event? Is that supported in the SDK?
> 
> In the prior implementation the intent was for the guardrail to clearly run and show the result, and the user could easily implement interruption/escalation/update response logic if it tripped as they'd like - what's the intent here?
> 

It comes from [the SDK](https://openai.github.io/openai-agents-js/guides/guardrails/) - now that these docs are more complete, I'll clean this up so that it demonstrates the original intent. I'll also make the guardrail distinct between NewTelco and Snowy Peak instead of sharing the same one.

bfioca-openai commented:
> Going through app.tsx is very confusing


Yup. Bringing back toward original design.

bfioca-openai commented:
> <img alt="image" width="268" src="https://private-user-images.githubusercontent.com/171723556/451263839-82140315-b53e-49bc-8b96-ca0fdc673a1d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkwNTI3NjEsIm5iZiI6MTc0OTA1MjQ2MSwicGF0aCI6Ii8xNzE3MjM1NTYvNDUxMjYzODM5LTgyMTQwMzE1LWI1M2UtNDliYy04Yjk2LWNhMGZkYzY3M2ExZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwNFQxNTU0MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Nzc5NzJmMTI0ZGQzYTZiMjk4YjNhMmUzNGUxZWI0MTdkMDMyYjY0ZTJlNzhlZTU3YjcyMWYzNGNkNTVjNjQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.nY9v-P6V8apnMf7cBTbQJbS6gn9YKksiN0_lmePvTmA">
> This UI also appears to be broken, is it doing anything in your testing? It should make the stream low bitrate and sound more like phone quality

Oversight - working on sorting out when a change to the SDK can be dropped to support, otherwise can either remove of monkeypatch.

bfioca-openai commented:
Reworked this quite a bit. Much easier now that the documentation is finalized. 

- Output guardrails are properly handled. 
- Codec works via monkey patch. 
- Restored `useHandleServerEvent.ts` and cleaned up App.tsx
- added `useRealtimeSession.ts` in place of the awkward wrapper Codex generated

nm-openai commented on src/app/agentConfigs/customerServiceRetail/index.ts:
```diff
@@ -17,3 +17,6 @@ export const customerServiceRetailScenario = [
 ];
+
+// Name of the company represented by this agent set. Used by guardrails
+export const customerServiceRetailCompanyName = 'Snowy Peak Boards';
```
This is better than before, but in the future I'd love this to just be "outputGuardrailCriteria", this is too bespoke to that one rule. And IMO if this isn't set we can not have output guardrails and if set we have it with those criteria added to the prompt

nm-openai reviewed (approved):
Mergeable thanks for making those changes!

bfioca-openai commented on src/app/hooks/useRealtimeSession.ts:
```diff
@@ -0,0 +1,209 @@
+
+  const { logServerEvent } = useEvent();
+
+  const historyHandlers = useHandleSessionHistory().current;
```
oh I like that! will fix in next patch!
